materialized_sql_view/model/abstract_materialized_sql_view.py

```diff
@@ -0,0 +1,344 @@
+            pg = PGMaterializedViewManager.getInstance(pg_version)
+            if not view_name:
+                view_name = self.sql_view_name
+            if not mat_view_name:
+                mat_view_name = self.sql_mat_view_name
+            pg.drop_mat_view(cr, view_name, mat_view_name)
+            self.after_drop_materialized_view(cr, uid, context=context)
+        except psycopg2.Error as e:
+            self.report_sql_error(cr, uid, e, pg_version, context=context)
+        else:
+            result = self.change_matview_state(cr, uid, 'after_drop_view',
+                                               pg_version, context=context)
+        return result
+
+    def report_sql_error(self, cr, uid, err, pg_version, context=None):
+        if not context:
+            context = {}
+        context.update({'error_message': err.pgerror})
+        cr.rollback()
+        self.change_matview_state(
+            cr, uid, 'aborted_matview', pg_version, context=context)
+
+    def before_drop_materialized_view(self, cr, uid, context=None):
+        """Method called before drop materialized view and view,
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def after_drop_materialized_view(self, cr, uid, context=None):
+        """Method called after drop materialized view and view,
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def before_create_materialized_view(self, cr, uid, context=None):
+        """Method called before create materialized view and view,
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def after_create_materialized_view(self, cr, uid, context=None):
+        """Method called after create materialized view and view,
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def before_refresh_materialized_view(self, cr, uid, context=None):
+        """Method called before refresh materialized view,
+           this was made to do things like drop index before in the same
+           transaction.
+
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def after_refresh_materialized_view(self, cr, uid, context=None):
+        """Method called after refresh materialized view,
+           this was made to do things like add index after refresh data
+
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def write(self, cr, uid, ids, context=None):
+        raise osv.except_osv(u"Write on materialized view is forbidden",
+                             u"Write on materialized view is forbidden,"
+                             u"because data would be lost at the next refresh"
+                             )
+
+    def create(self, cr, uid, ids, context=None):
+        raise osv.except_osv(u"Create data on materialized view is forbidden",
```

Comment:
osv.except_osv is deprecated use orm.except_orm

Don't forget to translate the strings